lib/travis/logs/helpers/database.rb

```diff
@@ -192,6 +192,14 @@ def delete_log_parts(log_id)
           db.call(:delete_log_parts, log_id: log_id)
         end
 
+        def log_parts(log_id, after: nil, part_numbers: [])
+          query = db[:log_parts].select(:id, :number, :content, :final)
+                                .where(log_id: log_id)
+          query = query.where('number > ?', after) if after
```

Comment:
If you defaulted after to 0 by doing a .to_i on params[:after] above would that screw anything up? I think you could to that and get rid of some of these nil set/checks.